specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json

```diff
@@ -4953,73 +4950,39 @@
       },
       "x-ms-discriminator-value": "AzureEventHubs"
     },
-    "AzureTableParameter": {
+    "AzureLogAnalyticsParameter": {
       "required": [
-        "connectionString",
+        "clientId",
```

Comment:
How will this work once you start supporting MSI?

Reply:
The related attributes will become optional if they choose MSI auth type for this data feed. And MSI will be used if they both exist.

Reply:
In general, you want to be explicit about how the service is going to authenticate to the external source. You may want to use a connection string in some cases even though the service runs as a specific identity. This is particularly true for services that may not support token auth today, but adds the capability at some point in the future (this is what azure storage did, for example). I would suggest making this a polymorphic type and add a `kind` property today, and that your service would honor this setting.

Reply:
If I understand this concern correctly, there are attributes [authenticationType](https://github.com/Azure/azure-rest-api-specs/blob/dee934c0578d310afc5072fdebf1b57e9180dc72/specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json#L4672), and credentialId in the DataFeedDetail model working with [DataSourceCredential](https://github.com/Azure/azure-rest-api-specs/blob/dee934c0578d310afc5072fdebf1b57e9180dc72/specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json#L4048). If MSI is used, the authenticationType will be ManagedIdentity, and if SerivcePrincipal is used, the authenticationType will be ServicePrincipal, and the parameters defined in given DataSourceCredential (created by user) will be used to auth.

Is this kind of strategy sufficient for the future? Or is there any quick adjustment that can be done to make it less possible to do a breaking change later?

Reply:
If we make the properties optional in the data source type, we should be good.

Reply:
Sure, double checked and removed some in the latest commit.